django/db/migrations/state.py

```diff
@@ -100,12 +100,23 @@ def reload_model(self, app_label, model_name):
 
             # Get all outgoing references from the model to be rendered
             model_state = self.models[(app_label, model_name)]
+            direct_related_models = set()
             for name, field in model_state.fields:
                 if field.is_relation:
                     if field.remote_field.model == RECURSIVE_RELATIONSHIP_CONSTANT:
                         continue
-                    rel_app_label, rel_model_name = _get_app_label_and_model_name(field.remote_field.model, app_label)
-                    related_models.add((rel_app_label, rel_model_name.lower()))
+                    rel_app_label, rel_model_name = _get_app_label_and_model_name(field.related_model, app_label)
+                    direct_related_models.add((rel_app_label, rel_model_name.lower()))
+
+            # For all direct related models, get all their recursive relations (#24573)
```

Comment:
maybe we could describe what "direct related models" are? not sure the ticket reference is helpful here
add period